src/sys/socket/addr.rs

```diff
@@ -973,4 +1052,97 @@ pub mod sys_control {
             write!(f, "id: {} unit: {}", self.id(), self.unit())
         }
     }
+
+    impl fmt::Debug for SysControlAddr {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            fmt::Display::fmt(self, f)
+        }
+    }
+}
+
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
+pub struct EtherAddr {
```

Comment:
This should be a new type around the underlying libc types, not a custom strict. See how the other types are declared for what time talking about. You'll just have to make them different definitions for the two different platforms types since the libc types are different.